WordPress/gutenberg
# Block Bindings: Move logic to core functions instead of using a hook

## What?
Move the bindings logic to get and update the value of the connected source to `getBlockAttributes` and `updateBlockAttributes` instead of having a custom hook.

## Why?
This is part of core and it is where it belongs. Apart from that, it ensure it works as expected in all the uses cases and not only when getting the attributes from the props and updating them with `setAttributes`.

## How?
* Remove the old hook implementation.
* As part of that, I moved a couple of utils to a new file and changed the imports of the necessary files.
* To change the attribute with the value returned by the source with `getValue`, hook into the `getBlockAttribute` selector to read the bindings.
* To allow editing of the source value through `setValue`, instead of updating the attribute itself, hook into the `updateBlockAttributes` action.

This last step requires changing the implementation to use `dispatch`, so I had to modify the e2e tests.

## Testing Instructions
e2e tests passing should be enough. If you want to test it manually you can:

<details>
<summary>1. Register post meta by adding this snippet to your theme's functions.php</summary>

```php
add_action( 'init', 'test_block_bindings' );

function test_block_bindings() {
	register_meta(
		'post',
		'text_field',
		array(
			'show_in_rest'      => true,
			'single'            => true,
			'type'              => 'string',
			'default'           => 'default text value',
		)
	);
}
```

</details>


<details>

<summary>2. Add a paragraph block bound to the custom field using the Code Editor</summary>

```

<p>Paragraph content</p>

```

</details>
 
3. Check that the value of the custom field is shown.
